asv/environment.py

```diff
@@ -444,7 +444,9 @@ def install_project(self, conf, commit_hash=None):
         if commit_hash is None:
             commit_hash = self._cache.get_existing_commit_hash()
             if commit_hash is None:
-                commit_hash = self.repo.get_hash_from_master()
+                commit_hash = self.repo.get_hash_from_name(conf.branches[0])
+                if commit_hash is None:
```

Comment:
get_hash_from_name will raise an exception if it can't resolve, so I think this fallback isn't needed.